components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/WebExtensionToolbarFeature.kt

```diff
@@ -94,6 +94,10 @@ class WebExtensionToolbarFeature(
     internal fun renderWebExtensionActions(state: BrowserState, tab: SessionState? = null) {
         val extensions = state.extensions.values.toList()
         extensions.filter { it.enabled }.sortedBy { it.name }.forEach { extension ->
+            if (!extension.allowedInPrivateBrowsing && tab?.content?.private == true) {
+                return
```

Comment:
Same as above. You don't want to return here, just continue with the next extension which could be allowed to run in private browsing.